Ouroboros/InfiniteCarousel.swift

```diff
@@ -190,19 +200,84 @@ public class InfiniteCarousel: UICollectionView, UICollectionViewDataSource, UIC
         
         jumping = false
         
-        let jumpDistance = CGFloat(count) * (collectionViewLayout as! Layout).totalItemWidth
-        let currentOffset = self.contentOffset.x
-        
         if focusHeading == .Left {
-            self.setContentOffset(CGPointMake(currentOffset + jumpDistance, self.contentOffset.y), animated: false)
+            jump(.Forward)
         } else {
-            self.setContentOffset(CGPointMake(currentOffset - jumpDistance, self.contentOffset.y), animated: false)
+            jump(.Backward)
         }
         
         currentlyFocusedItem = manualFocusCell!.item
         setNeedsFocusUpdate()
     }
     
+    func scrollToItem(item: Int, animated: Bool) {
+        if let initialOffset = (self.collectionViewLayout as! Layout).offsetForItemAtIndex(item) {
+            self.setContentOffset(CGPointMake(initialOffset,self.contentOffset.y), animated: animated)
+        }
+        
+        // Update focus element in case we have it
+        self.currentlyFocusedItem = item
+        self.manualFocusCell = NSIndexPath(forItem: self.currentlyFocusedItem, inSection: 0)
+        self.setNeedsFocusUpdate()
+    }
+    
+    // MARK: - Auto Scroll
+    
+    var scrollTimer: NSTimer?
+    var userInputTimer: NSTimer?
+    
+    func beginAutoScroll() {
+        guard autoScroll else {
+            return
+        }
+        
+        invalidateTimers()
+        scrollTimer = NSTimer.scheduledTimerWithTimeInterval(autoScrollTime, target: self,
+            selector: "scrollToNextPage", userInfo: nil, repeats: true)
+    }
+    
+    func pauseAutoScroll() {
+        guard autoScroll else {
+            return
+        }
+        
+        invalidateTimers()
+        userInputTimer = NSTimer.scheduledTimerWithTimeInterval(autoScrollReset, target: self,
```

Comment:
I agree with the assessment of just pausing forever rather than triggering a restart here, then checking in one of the updateFocusInContext methods to pause when focusing carousel and start when unfocusing.